deploy/contracts/Advertisement.sol

```diff
@@ -0,0 +1,92 @@
+pragma solidity ^0.4.8;
+/**
+ * The AdCampaign contract does this and that...
+ */
+contract Advertisement {
+
+	struct Filters {
+		string countries; 
+		string packageName;
+		uint[] vercodes;
+	}
+
+	struct ValidationRules {
+		bool vercode;
+		bool ipValidation;
+		bool country;
+		uint constipDailyConversions;
+		uint walletDailyConversions;
+	}
+
+	struct Campaign {	
+		bytes8 bid_id;
+		uint price;
+		uint budget;
+		uint startDate;
+		uint endDate;
+		string ipValidator;
+		address  owner;
+		Filters filters;
+	}
+
+	ValidationRules public rules;
+	mapping (bytes => Campaign[]) campaigns;
+
+	event CampaignCreated(string packageName, string countries, 
+							uint[] vercodes, uint price, uint budget,
+							uint startDate, uint endDate);
+
+	function Advertisement () public {
+		rules = ValidationRules(false,true,true,2,1);
+	}
+
+
+	function createCampaign (string packageName, string countries, 
```

Comment:
Does it make sense to make this a **external** method instead of **public** in order to avoid storing the inputs in memory and thus avoiding spending more gas? Do you think it will be called from other methods inside the contract?